### PR TITLE
Fix clang not producing debug info in cpp mode

### PR DESCRIPTION
**Pull request by nc-x**
Fixes https://github.com/nim-lang/Nim/issues/7957

* [EDIT by @timotheecour ] fixes https://github.com/nim-lang/Nim/issues/9973

### CODE REVIEW

**Comment by nc-x** [comment]
I have no  idea if
https://github.com/nim-lang/Nim/blob/77558e3183c8354857ff12980740a63a7e57fe77/config/nim.cfg#L229-L231
is required on OSes other than windows but it is required on windows for generating debug info.

File: config/nim.cfg [hunk 1]
```diff
@@ -220,10 +220,20 @@ llvm_gcc.options.size = "-Os"
...
 # Configuration for the LLVM CLang compiler:
 clang.options.debug = "-g"
+clang.cpp.options.debug = "-g"
 clang.options.always = "-w"
 clang.options.speed = "-O3"
 clang.options.size = "-Os"
 
+@if windows:
+  clang.options.debug = "-g -gcodeview"
+  clang.cpp.options.debug = "-g -gcodeview"
```
Comment (alaviss) [inline config/nim.cfg]: I think you can omit `-g`, because it's already been specified before this.

Comment (nc-x) [reply config/nim.cfg]: Nope. Assigning new value overwrites the previous one AFAIK.
The alternative is `clang.options.debug %= "${clang.options.debug} -gcodeview"`

**Comment by timotheecour** [comment]
also fixes https://github.com/nim-lang/Nim/issues/9973 ; yay!

**Pull request merged**